nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,867 @@
+"""High-level decoding object that exposes standard classification and
+regression strategies such as SVM, LogisticRegression and Ridge, with optional
+feature selection, integrated hyper-parameter selection and aggregation method.
+"""
+# Authors : Yannick Schwartz
+#           Andres Hoyos-Idrobo
+#           Binh Nguyen <tuan-binh.nguyen@inria.fr>
+#
+# License: simplified BSD
+
+import itertools
+import warnings
+
+import numpy as np
+from sklearn import clone
+from sklearn.base import RegressorMixin
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model.base import LinearModel
+from sklearn.linear_model.ridge import Ridge, RidgeClassifier, _BaseRidge
+from sklearn.model_selection import ParameterGrid, check_cv
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.svm import SVR, LinearSVC
+from sklearn.svm.bounds import l1_min_c
+from sklearn.utils.extmath import safe_sparse_dot
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+try:
+    from sklearn.externals.joblib import Parallel, delayed
+except ImportError:
+    # Scikit-learn will soon stop vendoring joblib
+    from joblib import Parallel, delayed
+
+try:
+    from sklearn.metrics import check_scoring
+except ImportError:
+    # for scikit-learn 0.18 and 0.19
+    from sklearn.metrics.scorer import check_scoring
+
+from .._utils import CacheMixin
+from .._utils.cache_mixin import _check_memory
+from .._utils.compat import _basestring
+from .._utils.param_validation import (_adjust_screening_percentile,
+                                       check_feature_screening)
+from ..input_data.masker_validation import check_embedded_nifti_masker
+
+SUPPORTED_ESTIMATORS = dict(
+    svc_l1=LinearSVC(penalty='l1', dual=False),
+    svc_l2=LinearSVC(penalty='l2'),
+    svc=LinearSVC(penalty='l2'),
+    logistic_l1=LogisticRegression(penalty='l1', solver='liblinear'),
+    logistic_l2=LogisticRegression(penalty='l2', solver='liblinear'),
+    logistic=LogisticRegression(penalty='l2', solver='liblinear'),
+    ridge_classifier=RidgeClassifier(),
+    ridge_regressor=Ridge(),
+    ridge=Ridge(),
+    svr=SVR(kernel='linear'),
+)
+
+
+def _check_param_grid(estimator, X, y, param_grid=None):
+    """Check param_grid and return sensible default if param_grid is None.
+
+    Parameters
+    -----------
+    estimator : str, optional
```

Comment:
The valid numpydoc standard requires no space before the ":"
```suggestion
    estimator: str, optional
```

Please change this for all occurrences of docstring parameters.